src/scripts/term_range_expansion.py

```diff
@@ -0,0 +1,207 @@
+from copy import copy
+from typing import Any, Dict, List
+
+from jinja2 import Template
+import pandas as pd
+
+
+# Vars
+# # Config
+USE_CACHE = False
+
+# # Static
+PROJECT_DIR = os.path.join(os.path.dirname(os.path.realpath(__file__)), '..')
+SPARQL_DIR = os.path.join(PROJECT_DIR, 'sparql')
+CACHE_DIR = os.path.join(PROJECT_DIR, 'cache')
+sparql_child_inclusion_path = os.path.join(PROJECT_DIR, 'sparql', 'get-terms-children.sparql.jinja2')
+sparql_child_exclusion_path = os.path.join(PROJECT_DIR, 'sparql', 'get-terms.sparql.jinja2')
+input_exclusion_term_ranges_csv_path = os.path.join(
+    PROJECT_DIR, 'preparation', 'icd10cm_excluded_term_ranges_intensional.csv')
+input_icd10cm_ttl_path = os.path.join(PROJECT_DIR, 'preparation', 'icd10cm.ttl')
+TERM_RANGE_MAX_VAL = 99
+
+
+# Functions
+# todo: pd.errors.EmptyDataError: I think this happens because Python process needs to stop before results saved?
+def sparql_jinja2_file_query__via_robot(
+    terms: List[str], query_template_path: str,
+    path_in_data_dir=os.path.basename(input_icd10cm_ttl_path), onto_path=input_icd10cm_ttl_path,
+    use_cache=USE_CACHE, include_superclass=False
+) -> pd.DataFrame:
+    """Query ontology using SPARQL query file"""
+    # Basic vars
+    query_template_filename = os.path.basename(query_template_path)
+    results_dirname = path_in_data_dir.replace('/', '-').replace('.', '-') + \
+        '__' + query_template_filename.replace('.', '-')
+    results_dirpath = os.path.join(CACHE_DIR, 'robot', results_dirname)
+    results_filename = 'results.csv'
+    command_save_filename = 'command.sh'
+    results_path = os.path.join(results_dirpath, results_filename)
+    command_save_path = os.path.join(results_dirpath, command_save_filename)
+    instantiated_query_path = os.path.join(results_dirpath, 'query.sparql')
+    command_str = f'robot query --input {onto_path} --query {instantiated_query_path} {results_path}'
+
+    # Instantiate template
+    with open(query_template_path, 'r') as f:
+        template_str = f.read()
+    template_obj = Template(template_str)
+    # https://www.w3.org/TR/sparql11-query/#propertypaths
+    instantiated_str = template_obj.render(
+        property_path_qualifier='*' if include_superclass else '+',
+        values=' '.join(terms))
+
+    # Cache and run
+    os.makedirs(results_dirpath, exist_ok=True)
+    if not (os.path.exists(results_path) and use_cache):
+        with open(instantiated_query_path, 'w') as f:
+            f.write(instantiated_str)
+        with open(command_save_path, 'w') as f:
+            f.write(command_str)
+        subprocess.run(command_str.split())
+
+    # Read results and return
+    try:
+        df = pd.read_csv(results_path).fillna('')
+    except pd.errors.EmptyDataError as err:
+        os.remove(results_path)  # done so that it doesn't read this from cache
+        raise err
+
+    return df
+
+
+def run(onto_path: str, exclusions_path: str) -> pd.DataFrame:
```

Comment:
This is still a work in progress. I'm repurposing this from the previous ICD10CM work.